testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestReadUniontype.java

```diff
@@ -139,6 +137,136 @@ public void testReadUniontype(String storageFormat)
         }
     }
 
+    @Test(dataProvider = "storage_formats", groups = SMOKE)
+    public void testUnionTypeSchemaEvolution(String storageFormat)
+    {
+        // According to testing results, the Hive INSERT queries here only work in Hive 1.2
+        if (getHiveVersionMajor() != 1 || getHiveVersionMinor() != 2) {
+            throw new SkipException("This test can only be run with Hive 1.2 (default config)");
+        }
+
+        onHive().executeQuery(format(
+                "CREATE TABLE %s ("
+                        + "c0 INT,"
+                        + "c1 UNIONTYPE<"
+                        + "     STRUCT<a:STRING, b:STRING>, "
+                        + "     STRUCT<c:STRING>>) "
+                        + "PARTITIONED BY (c2 INT) "
+                        + "STORED AS %s",
+                TABLE_NAME_SCHEMA_EVOLUTION,
+                storageFormat));
+
+        if (storageFormat.equals("AVRO")) {
```

Comment:
can you please convert this to switch case and throw exception for others

Reply:
Done:)